File: order/src/main/java/com/ecommerce/order/DTO/OrderResponse.java
package com.ecommerce.order.DTO;

import com.ecommerce.order.Enum.DeliveryStatus;
import com.ecommerce.order.Enum.PaymentStatus;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OrderResponse {

    private Integer orderId;
    private Double totalAmount;
    private Integer addressId;
    private LocalDateTime placedAt;

    private PaymentStatus paymentStatus;
    private DeliveryStatus deliveryStatus;

    private List<OrderItemResponse> items;
}

File: frontend/ecommerce/src/app/order/DTO/OrderResponse.ts
import {PaymentStatus} from '../Enum/PaymentStatus';
import {DeliveryStatus} from '../Enum/DelhiveryStatus';
import {OrderItemResponse} from './OrderItemResponse';

export interface OrderResponse{
  orderId:number;
  totalAmount:number;
  addressId:number;
  placedAt:Date;
  payentStatus:PaymentStatus;
  deliveryStatus:DeliveryStatus;
  orderItemResponse:OrderItemResponse[];


}
